tests/backend/test_handshake.py

```diff
@@ -229,17 +230,19 @@ async def test_handshake_expired_organization(backend, server_factory, expiredor
             root_verify_key=expiredorg.root_verify_key,
         )
 
-    async with server_factory(backend.handle_client) as server:
-        stream = server.connection_factory()
-        transport = await Transport.init_for_client(stream, server.addr.hostname)
+    with backend.event_bus.listen() as spy:
+        async with server_factory(backend.handle_client) as server:
+            stream = server.connection_factory()
+            transport = await Transport.init_for_client(stream, server.addr.hostname)
 
-        challenge_req = await transport.recv()
-        answer_req = ch.process_challenge_req(challenge_req)
+            challenge_req = await transport.recv()
+            answer_req = ch.process_challenge_req(challenge_req)
 
-        await transport.send(answer_req)
-        result_req = await transport.recv()
-        with pytest.raises(HandshakeOrganizationExpired):
-            ch.process_result_req(result_req)
+            await transport.send(answer_req)
+            result_req = await transport.recv()
+            with pytest.raises(HandshakeOrganizationExpired):
+                ch.process_result_req(result_req)
+    spy.assert_event_occured(BackendEvent.ORGANIZATION_EXPIRED)
```

Comment:
There is no guarantee the event will occur before we finish the connection, hence `spy.wait_with_timeout` should be used instead

Reply:
That's right, thanks